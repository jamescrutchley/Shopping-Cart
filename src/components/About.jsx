import aboutData from "../data/about.json";
import styles from "../styles/About.module.css";
import { processText } from "../helpers/processText";

import money from "../assets/icons/money-bag.svg";
import revenue from "../assets/icons/revenue.svg";
import target from "../assets/icons/target.svg";

const About = () => {
  const images = [revenue, money, target];

  return (
    <section className={styles.container}>
      {aboutData.map((section, index) => (
        <div key={index} className={styles.panel}>
          <img className={styles.image} src={images[index]} />
          <div className={styles.textContainer}>
            <h2 className={styles.heading}>{section.title}</h2>
            {processText(section.copy).map((para, index) => (
              <p key={index} className={styles.copy}>
                {para}
              </p>
            ))}
          </div>
        </div>
      ))}
    </section>
  );
};

export default About;
