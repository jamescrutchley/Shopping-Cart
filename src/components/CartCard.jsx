import styles from "../styles/CartCard.module.css";
import buttonStyles from "../styles/Buttons.module.css";
import { useEffect } from "react";
import ProductControls from "./ProductControls";
import QuantityControls from "./QuantityControls";

const CartCard = ({ data = null }) => {
  const { title, price, quantity, image } = data || {};

  return (
    <article className={styles.card}>
      <div className={styles.cardImg}>
        <img className={styles.thumbnail} src={image} alt="product thumbnail" />
      </div>
      <div className={styles.cardInfo}>
        <div className={styles.titleContainer}>
          <p>{title ?? ""}</p>
        </div>
        <div className={styles.priceContainer}>
          <p>${price}</p>
        </div>
        <QuantityControls className={styles.quantityControls} inCart={quantity} />
        <div className={styles.priceTotalContainer}>
          <p>${price * quantity}</p>
        </div>
        <div className={styles.productControlsContainer}></div>
      </div>
    </article>
  );
};

export default CartCard;
