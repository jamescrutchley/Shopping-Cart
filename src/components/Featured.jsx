import styles from "../styles/Featured.module.css";
import { Link } from "react-router-dom";

const Featured = () => {
  return (
    <div className={styles.featuredPanel}>
      <Link className={styles.link} to="/shop">
        <h2>Browse Our Collection</h2>
      </Link>
    </div>
  );
};

export default Featured;
