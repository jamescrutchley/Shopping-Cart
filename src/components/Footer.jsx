const Footer = () => {
    return (
        <footer>
            <p>Link to Github:</p>
        </footer>
    )
}

export default Footer;