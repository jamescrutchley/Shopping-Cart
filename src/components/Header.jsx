import styles from '../styles/Header.module.css'
import '../styles/Root.module.css'

const Header = () => {
    return (
        <header className={styles.header}>
            <h1>Product Store</h1>
        </header>
    )
}

export default Header;