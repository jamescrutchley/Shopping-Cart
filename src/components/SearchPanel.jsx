import styles from "../styles/SearchPanel.module.css";
import SearchButton from "./SearchButton";
import searchIcon from "../assets/icons/search.svg";
import buttonStyles from "../styles/Buttons.module.css";
import { useState } from "react";

const SearchPanel = ({ clickBack }) => {
  const [fadeOutClass, setFadeOutClass] = useState(null);

  // click 'outside' listener as well



  const notifyClickBack = () => {
    setFadeOutClass(true);
    setTimeout(() => {
      clickBack(null); // Call clickBack after 0.2s
    }, 150);
  };
  return (
    <section className={`${styles.panelWrapper} ${
        fadeOutClass ? styles.fade : ""
      }`}>
      <div
        className={`${styles.searchContainer} ${
          fadeOutClass ? styles.fade : ""
        }`}
      >
        <button onClick={() => notifyClickBack()} className={styles.back}>
          Back
        </button>
        <h2>Search Our Collection:</h2>

        <div className={styles.searchInputContainer}>
          <label>
            <p className={styles.visuallyHidden}>Search</p>
            <input type="text" />
          </label>
          <button className={buttonStyles.searchCollection}>
            <img src={searchIcon} alt="Search" />
          </button>
        </div>
      </div>
    </section>
  );
};

export default SearchPanel;
