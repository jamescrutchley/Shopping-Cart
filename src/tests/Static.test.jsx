import { render, screen } from "@testing-library/react";
import Header from "../components/Header.jsx";
import Footer from "../components/Footer.jsx";
import About from "../components/About.jsx";
import CartCard from "../components/CartCard.jsx";
import Featured from "../components/Featured.jsx";
import SearchPanel from "../components/SearchPanel.jsx"

import { describe, it, expect, vitest, vi } from "vitest";
import { BrowserRouter } from "react-router-dom";
import { fakeStore } from "../mocks/copiedObj.js";
import {renderWithCartContext, mockCartContext } from './mockContext/mockContext.jsx'

//wrapper for components that use 'cartcontext':


//Capturing and testing snapshots of static components.

describe("Static components render correctly.", () => {
    
  it("header snapshot", () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("footer snapshot", () => {
    const { asFragment } = render(<Footer />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("about snapshot", () => {
    const { asFragment } = render(<About />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("cartcard snapshot", () => {
    const { asFragment } = renderWithCartContext(
      <CartCard data={fakeStore[0]} />,
      mockCartContext
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("featured snapshot", () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Featured />
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it("searchpanel snapshot", () => {
    const { asFragment } = render(<SearchPanel />);
    expect(asFragment()).toMatchSnapshot();
  })
});
