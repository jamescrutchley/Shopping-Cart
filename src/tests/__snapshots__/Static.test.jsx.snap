// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Static components render correctly. > about snapshot 1`] = `
<DocumentFragment>
  <section
    class="_container_eec6d2"
  >
    <div
      class="_panel_eec6d2"
    >
      <img
        class="_image_eec6d2"
        src="/src/assets/icons/revenue.svg"
      />
      <div
        class="_textContainer_eec6d2"
      >
        <h2
          class="_heading_eec6d2"
        >
          dare to inspire
        </h2>
        <p
          class="_copy_eec6d2"
        >
          We take pride in our relentless pursuit of innovation, ensuring that our offerings stay at the forefront of technology and design.
        </p>
        <p
          class="_copy_eec6d2"
        >
          Our team of experts is always working tirelessly to bring you the best, keeping in mind affordability without compromising on quality. At Consumer Goods, we understand that your time is precious. That's why we've designed our shopping experience to be effortless and enjoyable.
        </p>
      </div>
    </div>
    <div
      class="_panel_eec6d2"
    >
      <img
        class="_image_eec6d2"
        src="/src/assets/icons/money-bag.svg"
      />
      <div
        class="_textContainer_eec6d2"
      >
        <h2
          class="_heading_eec6d2"
        >
          Timeless Innovation
        </h2>
        <p
          class="_copy_eec6d2"
        >
          We believe in creating products that stand the test of time while continually pushing the boundaries of what's possible. Our commitment to marrying tradition with cutting-edge technology results in innovations that endure, delivering lasting value and enriching the lives of those who embrace them. Each creation reflects our dedication to timelessness, ensuring that our legacy of innovation remains eternally relevant.
        </p>
      </div>
    </div>
    <div
      class="_panel_eec6d2"
    >
      <img
        class="_image_eec6d2"
        src="/src/assets/icons/target.svg"
      />
      <div
        class="_textContainer_eec6d2"
      >
        <h2
          class="_heading_eec6d2"
        >
          Immanentizing the Eschaton
        </h2>
        <p
          class="_copy_eec6d2"
        >
          Our relentless pursuit of excellence and innovation aims to create a future where every customer's dreams and desires are not just met but exceeded. We view each interaction as an opportunity to bring about a small slice of utopia in the form of exceptional products, unparalleled service, and unforgettable experiences.
        </p>
        <p
          class="_copy_eec6d2"
        >
          Our commitment to turning aspirations into reality is our way of making the extraordinary a part of everyday life, inching us closer to a world where the highest ideals are immanent and attainable through our steadfast pursuit of customer delight.
        </p>
      </div>
    </div>
  </section>
</DocumentFragment>
`;

exports[`Static components render correctly. > cartcard snapshot 1`] = `
<DocumentFragment>
  <article
    class="_card_d72dd6"
    data-testid="card"
  >
    <div
      class="_cardImg_d72dd6"
    >
      <img
        alt="product thumbnail"
        class="_thumbnail_d72dd6"
        src="https://fakestoreapi.com/img/81fPKd-2AYL._AC_SL1500_.jpg"
      />
    </div>
    <div
      class="_cardInfo_d72dd6"
    >
      <div
        class="_titleContainer_d72dd6"
      >
        <p>
          Fjallraven - Foldsack No. 1 Backpack, Fits 15 Laptops
        </p>
      </div>
      <div
        class="_priceContainer_d72dd6"
      >
        <p>
          $109.95
        </p>
      </div>
      <div
        class="_quantityControls_4bc385"
      >
        <label
          for="quantity"
        >
          Qty:
        </label>
        <div
          class="_controlsWrapper_4bc385"
        >
          <div
            class="_decrementContainer_4bc385"
          >
            <button>
              -
            </button>
          </div>
          <div
            class="_clickableInputContainer_4bc385"
          >
            <input
              class="_clickableInput_4bc385"
              id="quantity"
              max="9"
              min="1"
              name="quantity"
              type="quantity"
              value="4"
            />
          </div>
          <div
            class="_incrementContainer_4bc385"
          >
            <button>
              +
            </button>
          </div>
        </div>
      </div>
      <div
        class="_priceTotalContainer_d72dd6"
      >
        <p>
          $439.80
        </p>
      </div>
      <div
        class="_productControlsContainer_d72dd6"
      />
    </div>
  </article>
</DocumentFragment>
`;

exports[`Static components render correctly. > featured snapshot 1`] = `
<DocumentFragment>
  <div
    class="_featuredPanel_8c6ab7"
  >
    <a
      class="_link_8c6ab7"
      href="/shop"
    >
      <h2>
        Browse Our Collection
      </h2>
    </a>
  </div>
</DocumentFragment>
`;

exports[`Static components render correctly. > footer snapshot 1`] = `
<DocumentFragment>
  <footer
    class="_footer_4d8704"
  >
    <div
      class="_promotionContainer_4d8704"
    >
      <div
        class="_promotionInfo_4d8704"
      >
        <h3>
          Say Hello To Great Savings
        </h3>
        <p>
          <a
            href="#email"
          >
            Sign up now
          </a>
           to receive 10% off your first order!
        </p>
      </div>
      <div
        class="_signup_4d8704"
      >
        <label
          for="email"
        />
        <input
          id="email"
          placeholder="Your Email Here"
          type="text"
        />
      </div>
    </div>
    <div
      class="_nav_4d8704"
    >
      <div
        class="_navLinks_4d8704"
      >
        <div
          class="_linksGroup_4d8704"
        >
          <h5>
            Information
          </h5>
          <ul
            class="_linksList_4d8704"
          >
            <li>
              Our History
            </li>
            <li>
              Delivery
            </li>
            <li>
              Brands
            </li>
          </ul>
        </div>
        <div
          class="_linksGroup_4d8704"
        >
          <h5>
            Services
          </h5>
          <ul
            class="_linksList_4d8704"
          >
            <li>
              Recalls
            </li>
            <li>
              Magazine
            </li>
            <li>
              Subscription
            </li>
          </ul>
        </div>
        <div
          class="_linksGroup_4d8704"
        >
          <h5>
            Our Vision
          </h5>
          <ul
            class="_linksList_4d8704"
          >
            <li>
              Your Account
            </li>
            <li>
              Track Your Order
            </li>
            <li>
              Returns and Refunds
            </li>
          </ul>
        </div>
      </div>
      <div
        class="_divider_4d8704"
      />
      <div
        class="_contact_4d8704"
      >
        <div
          class="_contactInfo_4d8704"
        >
          <h4>
            Get In Touch
          </h4>
          <p>
            Our friendly staff are here to help. In-store, online, anytime!
          </p>
        </div>
        <ul>
          <li>
            Phone
          </li>
          <li>
            Email
          </li>
          <li>
            Live Chat
          </li>
        </ul>
      </div>
    </div>
    <div
      class="_legal_4d8704"
    >
      <ul>
        <li>
          Policies
        </li>
        <li>
          Terms and Conditions
        </li>
        <li>
          For Suppliers
        </li>
        <li>
          About Us
        </li>
      </ul>
    </div>
    <div
      class="_subFooter_4d8704"
    >
      <a
        href="https://github.com/jamescrutchley"
      >
        <img
          alt="github logo"
          src="/src/assets/icons/github-mark.svg"
        />
      </a>
      <p>
        Developed by James Crutchley
      </p>
    </div>
  </footer>
</DocumentFragment>
`;

exports[`Static components render correctly. > header snapshot 1`] = `
<DocumentFragment>
  <header
    class="_header_a9d33c"
  >
    <a
      href="/"
    >
      <h1>
        Consumer Goods
      </h1>
    </a>
  </header>
</DocumentFragment>
`;

exports[`Static components render correctly. > searchpanel snapshot 1`] = `
<DocumentFragment>
  <section
    class="_panelWrapper_7f3fc5 "
  >
    <div
      class="_searchContainer_7f3fc5 "
    >
      <button
        class="_back_7f3fc5"
      >
        Back
      </button>
      <h2>
        Search Our Collection:
      </h2>
      <div
        class="_searchInputContainer_7f3fc5"
      >
        <label>
          <p
            class="_visuallyHidden_7f3fc5"
          >
            Search
          </p>
          <input
            type="text"
          />
        </label>
        <button
          class="_searchCollection_4bc385"
        >
          <img
            alt="Search"
            src="/src/assets/icons/search.svg"
          />
        </button>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
